pkg/eventcounter/api/api.go

```diff
@@ -289,104 +291,124 @@ func (s *eventCounterService) GetEvaluationTimeseriesCount(
 	}
 	hourlyTimeStamps := getHourlyTimeStamps(timestamps, timestampUnit)
 	vIDs := getVariationIDs(resp.Feature.Variations)
-	variationTSEvents := make([]*ecproto.VariationTimeseries, 0, len(vIDs))
-	variationTSUsers := make([]*ecproto.VariationTimeseries, 0, len(vIDs))
-	for _, vID := range vIDs {
-		eventCountKeys := s.getEventCountKeys(
-			hourlyTimeStamps,
-			req.EnvironmentId,
-			req.FeatureId,
-			vID,
-		)
-		userCountKeys := s.getUserCountKeys(
-			hourlyTimeStamps,
-			req.EnvironmentId,
-			req.FeatureId,
-			vID,
-		)
-		eventCounts, err := s.getEventCounts(eventCountKeys, timestampUnit)
-		if err != nil {
-			s.logCountError(
-				ctx,
-				err,
-				"Failed to get event counts", req.EnvironmentId, req.FeatureId, vID,
-				resp.Feature.Version,
-				timestampUnit, req.TimeRange,
+
+	variationEventCountsMap := make(map[string][]float64, len(vIDs))
+	variationUserCountsMap := make(map[string][]float64, len(vIDs))
+	variationTotalEventCountsMap := make(map[string]int64, len(vIDs))
+	variationTotalUserCountsMap := make(map[string]int64, len(vIDs))
+
+	var mu sync.Mutex
+	var eg errgroup.Group
+
+	for _, variationID := range vIDs {
+		eg.Go(func() error {
```

Comment:
By using the `variationID` directly from the loop, depending on the time, the `variationID` will be overridden by the loop, and the functions will get the wrong counter when making requests to Redis.

```suggestion
	for _, vID := range vIDs {
	        variationID := vID
		eg.Go(func() error {
```

Reply:
Oh, this problem was fixed in Go1.22(https://go.dev/blog/loopvar-preview).
From Go1.22,  each iteration of the for loop gets its own copy of the loop variable. So we don't need redeclare the variable!